src/attached_probe.cpp

```diff
@@ -691,10 +691,19 @@ void AttachedProbe::load_prog()
     if (bt_verbose)
       log_level = 1;
 
-    // Use orig_name for program name so we get proper name for
-    // wildcard probes, replace wildcards with '.'
-    name = probe_.orig_name;
-    std::replace(name.begin(), name.end(), '*', '.');
+    if (probe_.type == ProbeType::kfunc || probe_.type == ProbeType::kretfunc)
+    {
+      // Use attach_point for program name when probe type is kfunc or
+      // kretfunc, otherwise will get a ENOENT when using a wildcard
+      name = probe_.attach_point;
+    }
+    else
+    {
+      // Use orig_name for program name so we get proper name for
+      // wildcard probes, replace wildcards with '.'
+      name = probe_.orig_name;
+      std::replace(name.begin(), name.end(), '*', '.');
+    }
```

Comment:
Since the commit that caused this is 2b7e38ab60bf288ed8681ae08135710b5f5f6b1e, I'd suggest something like:
```suggestion
    if (probe_.type == ProbeType::kprobe || probe_.type == ProbeType::kretprobe)
    {
        // Use orig_name for program name so we get proper name for
        // wildcard probes, replace wildcards with '.'
        name = probe_.orig_name;
        std::replace(name.begin(), name.end(), '*', '.');
    }
    else
        name = probe_.name
```

Reply:
Yeah, only use the orig_name for kprobe and kretprobe, this is better.